### PR TITLE
Fix @-file mentioning

### LINKED ISSUE #1045: JetBrains: Crash occurred with @-file tagging "java.lang.UnsupportedOperationException: JsonNull"
### Installation Information

Plugin version: ```5.3.4811-nightly```
IDE version: ```IC-233.14015.106```

### Describe the bug

**Steps to reproduce:**

1. Have Cody plugin installed in your JetBrains IntelliJ IDEA editor
2. User is already logged in to Cody with Free/Pro/Enterprise account
3. Go to Cody Chat UI and ask a question with @-file tagging
4. i.e. explain code from this file @filename.py

### Observed behavior:
Crash occurred with following stacktrace,

Stacktrace:
```text
java.lang.UnsupportedOperationException: JsonNull
	at com.google.gson.JsonElement.getAsString(JsonElement.java:187)
	at com.sourcegraph.cody.agent.protocol.ContextFileKt.contextFileDeserializer$lambda$0(ContextFile.kt:62)
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:76)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.CollectionTypeAdapter.read(CollectionTypeAdapter.java:114)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.CollectionTypeAdapter.read(CollectionTypeAdapter.java:40)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.readIntoField(ReflectiveTypeAdapterFactory.java:212)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$FieldReflectionAdapter.readField(ReflectiveTypeAdapterFactory.java:433)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:393)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.CollectionTypeAdapter.read(CollectionTypeAdapter.java:114)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.CollectionTypeAdapter.read(CollectionTypeAdapter.java:40)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.readIntoField(ReflectiveTypeAdapterFactory.java:212)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$FieldReflectionAdapter.readField(ReflectiveTypeAdapterFactory.java:433)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:393)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.readIntoField(ReflectiveTypeAdapterFactory.java:212)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$FieldReflectionAdapter.readField(ReflectiveTypeAdapterFactory.java:433)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:393)
	at com.google.gson.Gson.fromJson(Gson.java:1227)
	at com.google.gson.Gson.fromJson(Gson.java:1186)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.fromJson(MessageTypeAdapter.java:345)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.parseParams(MessageTypeAdapter.java:265)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.read(MessageTypeAdapter.java:121)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.read(MessageTypeAdapter.java:57)
	at com.google.gson.Gson.fromJson(Gson.java:1227)
	at com.google.gson.Gson.fromJson(Gson.java:1186)
	at org.eclipse.lsp4j.jsonrpc.json.MessageJsonHandler.parseMessage(MessageJsonHandler.java:119)
	at org.eclipse.lsp4j.jsonrpc.json.MessageJsonHandler.parseMessage(MessageJsonHandler.java:114)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:193)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)

```

### Expected behavior

Crash should no occur and Cody should explain code from that file.

### Additional context

**Video link:**

https://drive.google.com/file/d/13OcEfEXlv1-WQGpu25w5Nj9G3F7eoDM6/view?usp=drive_link

### PR DESCRIPTION
**Pull request by mkondratek**
Fixes https://github.com/sourcegraph/jetbrains/issues/1045.

There were 2 issues here:
1. We were setting model to null for enterprise accounts - that caused an error in the agent
2. We were not adjusted to the latest changes to the agent related to the context files (https://github.com/sourcegraph/cody/pull/3269)

## Test plan
1. @-file works

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/ContextItem.kt [hunk 1]
```diff
@@ -1,25 +1,51 @@
...
+    override val type: String,
     override val uri: URI,
-    override val repoName: String?,
-    override val revision: String?,
     val range: Range? = null,
-) : ContextFile() {
-  override val type: String = "file"
+    val repoName: String? = null,
+    val revision: String? = null,
+    val title: String? = null,
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/protocol/ContextItem.kt:43]: for now large files have title `large-file`

File: src/main/kotlin/com/sourcegraph/cody/chat/ui/ChatPanel.kt [hunk 2]
```diff
@@ -116,7 +116,7 @@ class ChatPanel(
   private fun setLlmForAgentSession(chatModelProvider: ChatModelsResponse.ChatModelProvider) {
     val activeAccountType = CodyAuthenticationManager.instance.getActiveAccount(project)
     if (activeAccountType.isEnterpriseAccount()) {
-      chatSession.sendWebviewMessage(WebviewMessage(command = "chatModel"))
+      // no need to send the webview message since the chat model is set by default
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/chat/ui/ChatPanel.kt:119]: this is important part of the fix

File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/ContextItem.kt [hunk 3]
```diff
@@ -1,25 +1,51 @@
...
-  abstract val type: String
+typealias ContextFileSource =
+    String // One of: embeddings, user, keyword, editor, filename, search, unified, selection,
+// terminal
+
+typealias SymbolKind = String // One of: class, function, method
+
+sealed class ContextItem {
+  abstract val type: String // Oneof: file, symbol
   abstract val uri: URI
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/protocol/ContextItem.kt:22]: The generated protocol is not perfect. In particular `ContextItem` contains no fields in it while these two can and should be included here. 

I think that in general the binding script and the generated binding are helpful but we must carefully review them.

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**